src/atopile/kicad_plugin/pushgroup.py

```diff
@@ -48,15 +55,29 @@ def Run(self):
             for item in target_board.GetTracks():
                 target_board.Remove(item)
 
+            # Calculate offset before moving footprints
+            offset = calculate_translation(source_fps=get_group_footprints(g), target_fps=source_board.GetFootprints())
+
             # Push the layout
             sync_footprints(
                 source_board, target_board, known_layouts[g_name]["uuid_map"]
             )
+
             for track in g.GetItems():
                 if isinstance(track, pcbnew.PCB_TRACK):
-                    sync_track(track, target_board)
+                    sync_track(source_board, track, target_board)
+
+            # Shift all objects of interest by offset
+            for fp in target_board.GetFootprints():
+                fp.Move(offset)
+
+            for track in target_board.GetTracks():
+                track.Move(offset)
 
-        # Save the target board
-        target_board.Save(target_board.GetFileName())
+            # Save the target board
```

Comment:
I indented this so it saved every target board you might've been updating, rather than just the first